doc/liquidsoap.xml

```diff
@@ -26,6 +26,11 @@
       <item>then</item>
       <item>else</item>
       <item>elsif</item>
+      <item>while</item>
+      <item>for</item>
+      <item>to</item>
+      <item>do</item>
+      <item>done</item>
```

Comment:
When working on the syntax for errors, I realized that we should be careful introducing too many reserved keywords. I was trying to use `in` for `try ... catch err in [errs] do ... end` and `in` was already wildly used (`in()`, `fade.in` etc.)

At the very least we should re-use `end` instead of `done` and perhaps also refrain from adding too many intermediary words, for instance:
```
for n = 1 .. n do
  [...]
end
```

Another thing I noticed reading from `go` is that they re-use `for` for `while` loops. I think I prefer `while` but just wanted to note it.

Reply:
Alright, I changed the `do` to `end`. I am not fond of the `..` syntax: I would be alright to add it if we had proper iterators (which we don't for now :))

Reply:
Cool. I was just thinking, why don't we make the iteration more functional and generic? Something like:
```
for n = array do
   (...)
end
```

Then add a syntax for:
```
x = 1..5 // [1, 2, 3, 4, 5]
```

I'd keep the syntax for brackets away because we might want to do this at some point:
```
x = [2,3]
x = [1, ...x] // [1,2,3]
```

I think `1..5` syntax would be useful to our users as well.

Reply:
I had this in mind too, I was lazy but since you propose that I'll dive into it. I propose
- to add a constraint `Iterable` for iterable types (lists, etc.)
- the syntax should be `for i in l do ... end` (I'd rather have `in` than `=`)
- I don't like the idea of building a list, I think that `1..1000` would generate something like [Seq.t](https://caml.inria.fr/pub/docs/manual-ocaml/libref/Seq.html) in OCaml, i.e. a function of type `() -> int?` which returns the integers from 1 to 1000 and then `null` (this avoids uselessly building the list)
Do you like it?

Reply:
I'm down with the following remarks:
* Do we really need `iterable`? Couldn't we just start with `() -> 'a?` for `for` and `() -> true` for while?
* I have nothing against using `in` per-say but what will you do with `fade.in`?

Reply:
Yes we could start with `() -> 'a?` but it would be nice to be able to iterate over a list without having to convert it first...

I think I should be able to keep `fade.in` with some easy parser tweaking.